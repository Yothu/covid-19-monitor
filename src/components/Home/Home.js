import { useEffect } from 'react';
import styled from 'styled-components';
import { useDispatch, useSelector } from 'react-redux';
import { getRegionsAndTotals, listRegions } from '../../redux/regions/regions';
import Country from '../Country/Country';
import Regions from '../Regions/Regions';

const Container = styled.div``;

const Home = () => {
  const dispatch = useDispatch();
  const { regions } = useSelector((state) => state.regions);

  useEffect(async () => {
    if (regions.length === 0) {
      const APIData = await getRegionsAndTotals();
      dispatch(listRegions(APIData));
    }
  }, []);

  return (
    <Container>
      <Country />
      <Regions />
    </Container>
  );
};

export default Home;
