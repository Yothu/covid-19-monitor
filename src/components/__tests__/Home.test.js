import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import Home from '../Home/Home';
import store from '../../redux/configureStore';

describe('Home page tests', () => {
  it('it renders correctly', () => {
    const home = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Home />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(home).toMatchSnapshot();
  });
});
