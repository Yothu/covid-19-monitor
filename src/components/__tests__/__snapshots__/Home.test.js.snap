// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page tests it renders correctly 1`] = `
<div
  className="sc-ksdxgE gagolF"
>
  <section
    className="sc-bdvvtL jsqFpi"
  >
    <div
      className="sc-gsDKAQ YHamg"
    >
      <img
        alt="chile-silhouette"
        src="chile-silhouette.svg"
        style={
          Object {
            "filter": "invert(100)",
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="sc-dkPtRN fASbqE"
    >
      <h2
        className="sc-hKwDye felzUJ"
      />
      <p
        className="sc-eCImPb ehWAiM"
      >
        <span
          className="sc-jRQBWg iHFUri"
        >
          new confirmed
        </span>
      </p>
      <p
        className="sc-eCImPb ehWAiM"
      >
        <span
          className="sc-jRQBWg iHFUri"
        >
          new deaths
        </span>
      </p>
      <p
        className="sc-eCImPb ehWAiM"
      >
        <span
          className="sc-jRQBWg iHFUri"
        >
          deaths
        </span>
      </p>
      <p
        className="sc-eCImPb ehWAiM"
      >
        <span
          className="sc-jRQBWg iHFUri"
        >
          confirmed
        </span>
      </p>
    </div>
  </section>
  <section
    className="sc-kDTinF eKdzuN"
  >
    <div
      className="sc-iqseJM fcRNMj"
    >
      Stats by Region
      <div
        className="sc-egiyK rWYOo"
      >
        Filtered by:
      </div>
    </div>
    <ul
      className="sc-crHmcD KbYQv"
    />
  </section>
</div>
`;
